app/code/Magento/Quote/Observer/Frontend/Quote/Address/CollectTotalsObserver.php

```diff
@@ -131,7 +143,12 @@ public function execute(\Magento\Framework\Event\Observer $observer)
             );
         }
 
-        if ($groupId !== null) {
+        // Do not update customer group as not logged in when doing process in admin area
+        if ($groupId !== null
+            && !(
+                $this->state->getAreaCode() == Area::AREA_ADMINHTML
```

Comment:
Please use a strict comparison operator `===` and add `declare(strict_types=1);`